kythe/cxx/common/indexing/frontend.cc

```diff
@@ -137,18 +137,21 @@ void DecodeIndexFile(const std::string& path,
   close(fd);
 }
 
-/// \brief Reads a single compilation from a .kzip file into memory.
+/// \brief Reads all compilations from a .kzip file into memory.
 /// \param path The path from which the file should be read.
-/// \param virtual_files A vector to be filled with FileData.
-/// \param unit A `CompilationUnit` to be decoded from the .kzip.
-void DecodeKZipFile(const std::string& path,
-                    std::vector<proto::FileData>* virtual_files,
-                    proto::CompilationUnit* unit) {
+/// \param jobs A vector to add a job to for each compilation in the kzip.
+/// \param silent The silent flag is copied to each of the jobs created from the
+/// kzip file.
+void DecodeKZipFile(const std::string& path, std::vector<IndexerJob>* jobs,
+                    bool silent) {
   StatusOr<IndexReader> reader = kythe::KzipReader::Open(path);
   CHECK(reader) << "Couldn't open kzip from " << path;
   bool compilation_read = false;
   auto status = reader->Scan([&](absl::string_view digest) {
-    CHECK(!compilation_read) << "Found more than 1 compilation in " << path;
+    jobs->emplace_back();
```

Comment:
Is it easy/possible for the indexer to only have a single compilation unit in memory at a time?  If not, this should probably be a `TODO`.

Reply:
Should definitely be possible, but probably would require a fair amount of refactoring. Right now the `IndexerFrontend` builds a list of "jobs" from the specified files/units, then the list is later consumed by the indexers.

The other languages just call `visit()` for each compilation unit found, which seems like a good way to avoid storing all units at once during the process.